lib/fragment.js

```diff
@@ -96,22 +103,29 @@ module.exports = class Fragment extends EventEmitter {
 
         const spanOptions = parentSpan ? { childOf: parentSpan } : {};
         const span = tracer.startSpan('fetch-fragment', spanOptions);
+
+        const {
+            id = 'unnamed',
+            primary,
+            async: isAsync,
+            public: isPublic
+        } = this.attributes;
+
         span.addTags({
             [Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_CLIENT,
             [Tags.HTTP_URL]: url,
-            id: this.attributes.id || 'unnamed',
             fallback: isFallback,
-            primary: this.attributes.primary,
-            async: this.attributes.async,
-            public: this.attributes.public
+            public: isPublic,
+            async: isAsync,
+            id,
```

Comment:
The behaviour for id has changed, default arguments only apply to undefined values. Null or empty string will passthrough.

Reply:
good catch. changing it back